### PR TITLE
Add retries to FakeTransactionalJobQueue

### PR DESCRIPTION
**Pull request by drashkov**
Adds the ability to test jobs failing and restarting.

### CODE REVIEW

File: misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeTransactionalJobQueue.kt [hunk 1]
```diff
@@ -73,15 +75,16 @@ class FakeTransactionalJobQueue @Inject constructor(
...
+    retries: Int = 1
   ): List<FakeJob> {
     val jobHandler = jobHandlers.get()[queueName]!!
     val jobs = jobQueues[queueName] ?: return listOf()
 
     val result = mutableListOf<FakeJob>()
     while (true) {
       val job = jobs.poll() ?: break
-      jobHandler.handleJob(job)
+      retry(retries,  FlatBackoff(Duration.ofMillis(100))) { jobHandler.handleJob(job) }
```
Comment (alanpaulin) [inline misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeTransactionalJobQueue.kt:87]: ultranit: there are 2 spaces after `retries,`

Comment (alanpaulin) [inline misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeTransactionalJobQueue.kt:87]: Just curious, where did the constant `100` come from? Is that what is used elsewhere?

File: misk-jobqueue-testing/src/test/kotlin/misk/jobqueue/FakeTransactionalJobQueueTest.kt [hunk 2]
```diff
@@ -215,6 +240,17 @@ private class Starport @Inject private constructor(moshi: Moshi) : JobHandler {
   }
 }
 
+private class Crasher @Inject private constructor(val tokenGenerator: TokenGenerator) : JobHandler {
+
+  override fun handleJob(job: Job) {
+    // Tokens are generated once per test run, starting from 1.
+    if (tokenGenerator.generate().endsWith("1")) {
+      throw UnitException()
```
Comment (alanpaulin) [inline misk-jobqueue-testing/src/test/kotlin/misk/jobqueue/FakeTransactionalJobQueueTest.kt:248]: Just curious: Is there a significance to `UnitException`, or could it be any `Exception` class?

**Review by alanpaulin — APPROVED** [review]
LGTM

File: misk-jobqueue-testing/src/test/kotlin/misk/jobqueue/FakeTransactionalJobQueueTest.kt (hunk 2 above)
Comment (drashkov) [reply misk-jobqueue-testing/src/test/kotlin/misk/jobqueue/FakeTransactionalJobQueueTest.kt:248]: Could be any checked exception, this one is keeping with the starcraft theme of this test class.

File: misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeTransactionalJobQueue.kt (hunk 1 above)
Comment (drashkov) [reply misk-jobqueue-testing/src/main/kotlin/misk/jobqueue/FakeTransactionalJobQueue.kt:87]: Pretty arbitrary, but small timeouts like this are used in the other tests.

**Pull request merged**